### ISSUE TITLE
Questions: About the config for drivaernet++

### ISSUE THREAD

**Issue by HydrogenSulfate** [Shizheng-Wen/GAOT-3D#7]
First, thank you very much for open-sourcing this excellent work. While attempting to reproduce the results using the provided pressure.yaml, I noticed that the value of `model.args.latent_tokens` in the config differs slightly from what is described in the paper.

<https://github.com/Shizheng-Wen/GAOT-3D/blob/56257d1166f55ed14a7189ade307bd7f4efa2eea/config/examples/drivaernet/pressure.yaml#L30>

<img width="1234" height="398" alt="Image" src="https://github.com/user-attachments/assets/3ceb1d05-d0d1-4d6f-8bf2-09503b146888" />

May I ask what the reason for this discrepancy is? Additionally, when reproducing the results with the `pressure.yaml`, the accuracy I obtained (4× A100 40GB) is slightly lower than what was reported in the paper. I am wondering whether this difference might be caused by the mismatch in latent_tokens between the config and the paper.

<img width="732" height="617" alt="Image" src="https://github.com/user-attachments/assets/08d2d3aa-cd41-4923-8bf6-be6353f3eaaf" />

<img width="586" height="361" alt="Image" src="https://github.com/user-attachments/assets/9ecf50a8-45bb-4cc1-8d36-4846870e7ee7" />

Thanks in advance for your clarification.

**Comment by Shizheng-Wen** [comment]
Thanks for pointing that out! The value on the paper is wrong. It should be the value in the pressure.yaml. I will fix it in the next arXiv version.

As for the discrepancy, this is because of the learning rate setting in the config files. I just realized that I uploaded a version with a too small value.  I have fixed it now. Please use this new one! For your reference, I have attached the content of the log files:

Btw, I haven't done any parameter tuning for training the model. So I believe you can get better results than my paper if you try some new setup. 

Please always feel free to reach out to me if you have any questions! I am always happy to help.

Best,
Shizheng

```
2025-10-15 20:28:23 2025-10-15 22:28:23 INFO src.trainer.stat: Initializing PyG dataset ...
2025-10-15 20:28:23 2025-10-15 22:28:23 INFO src.trainer.stat: Rank 0: Dataset initialization finished.
2025-10-15 20:28:23 2025-10-15 22:28:23 INFO src.trainer.stat: Loading pre-calculated normalization stats from /iopsstor/scratch/cscs/wshizhen/graphnpde/drivaernet/drivaernet_fullpressure_norm_stats.pt
2025-10-15 20:28:23 2025-10-15 22:28:23 INFO src.trainer.stat: Loaded x - Mean: tensor([-93.4105]), Std: tensor([120.7879])
2025-10-15 20:28:23 2025-10-15 22:28:23 INFO src.trainer.stat: Loaded c - Mean: tensor([7.3248e-03, 4.8269e-05, 5.7993e-03]), Std: tensor([0.4669, 0.6076, 0.6424])
2025-10-15 20:28:23 2025-10-15 22:28:23 INFO src.trainer.stat: Rank 0: Loading train dataset...
2025-10-15 20:28:24 Loaded 5817 samples for split 'train'.
2025-10-15 20:28:24 2025-10-15 22:28:24 INFO src.trainer.stat: Rank 0: Loading validation dataset...
2025-10-15 20:28:24 Loaded 1148 samples for split 'val'.
2025-10-15 20:28:24 2025-10-15 22:28:24 INFO src.trainer.stat: Rank 0: Loading test dataset...
2025-10-15 20:28:24 Loaded 1154 samples for split 'test'.
2025-10-15 20:28:25 2025-10-15 22:28:25 INFO src.trainer.stat: Rank 0: Created DistributedSampler for training.
2025-10-15 20:28:25 2025-10-15 22:28:25 INFO src.trainer.stat: Rank 0: Created DistributedSampler for validation.
2025-10-15 20:28:25 2025-10-15 22:28:25 INFO src.trainer.base: Number of parameters: 11244737
2025-10-15 20:28:25 AdamWOptimizer: 100%|██████████| 50/50 [21:14:03<00:00, 1528.88s/it, loss=0.0352, val_loss=0.17]    
2025-10-16 17:42:29 /users/wshizhen/gaot3d-private/src/trainer/base.py:402: UserWarning: No artists with labels found to put in legend.  Note that artists whose label start with an underscore are ignored when legend() is called with no argument.
2025-10-16 17:42:29   ax[1].legend()
2025-10-16 17:42:31 2025-10-16 19:42:31 INFO src.trainer.base: Saved checkpoint artifact: alps-gaot-drivaernet-pressure-normals-ps2-10blocks-geo_ngeo:['latest', 'best']
2025-10-16 17:42:31 2025-10-16 19:42:31 INFO src.trainer.stat: Starting testing with metric suite: 'drivaernet'
2025-10-16 17:42:32 2025-10-16 19:42:32 INFO src.trainer.stat: Extracted plotting data of N_S_WWS_WM_172:coords shape (525290, 3), gtr shape (525290, 1), prd shape (525290, 1)
2025-10-16 17:49:16 2025-10-16 19:49:16 INFO src.trainer.stat: Concatenated results: preds shape torch.Size([516881608, 1]), targets shape torch.Size([516881608, 1])
2025-10-16 17:49:28 2025-10-16 19:49:28 INFO src.trainer.stat: --- Final Metrics (Drivaernet Suite - Full Dataset) ---
2025-10-16 17:49:28 2025-10-16 19:49:28 INFO src.trainer.stat: MSE (x10^-2):       4.2444
2025-10-16 17:49:28 2025-10-16 19:49:28 INFO src.trainer.stat: MAE (x10^-1):       1.0727
2025-10-16 17:49:28 2025-10-16 19:49:28 INFO src.trainer.stat: RMSE:               0.1975
2025-10-16 17:49:28 2025-10-16 19:49:28 INFO src.trainer.stat: Max_Error:          9.4222
2025-10-16 17:49:28 2025-10-16 19:49:28 INFO src.trainer.stat: Rel L2 Error (%):   19.9395
2025-10-16 17:49:28 2025-10-16 19:49:28 INFO src.trainer.stat: Rel L1 Error (%):   18.0177
2025-10-16 17:49:28 2025-10-16 19:49:28 INFO src.trainer.stat: Attempting to plot first sample...
2025-10-16 17:49:28 /users/wshizhen/gaot3d-private/src/trainer/utils/plot.py:285: UserWarning: This figure includes Axes that are not compatible with tight_layout, so results might be incorrect.
2025-10-16 17:49:28   plt.tight_layout()
2025-10-16 17:49:42 Saved 3D comparison plot (matplotlib) to /users/wshizhen/gaot3d-private/.results/drivaernet/crv/gaot/pressure_normals_ps2_10blocks_geo_ngeo.png
2025-10-16 17:49:42 2025-10-16 19:49:42 INFO src.trainer.stat: Saved plot for first test sample to /users/wshizhen/gaot3d-private/.results/drivaernet/crv/gaot/pressure_normals_ps2_10blocks_geo_ngeo.png
```

**Comment by HydrogenSulfate** [comment]
> Thanks for pointing that out! The value on the paper is wrong. It should be the value in the pressure.yaml. I will fix it in the next arXiv version.
> 
> As for the discrepancy, this is because of the learning rate setting in the config files. I just realized that I uploaded a version with a too small value. I have fixed it now. Please use this new one! For your reference, I have attached the content of the log files:
> 
> Btw, I haven't done any parameter tuning for training the model. So I believe you can get better results than my paper if you try some new setup.
> 
> Please always feel free to reach out to me if you have any questions! I am always happy to help.
> 
> Best, Shizheng

Thank you very much for sharing the updated logs and the corrected code. I really appreciate your detailed explanation and the fix.

I will retrain and test the model based on your suggestions and the new configuration.
Thanks again for your help!

**Comment by HydrogenSulfate** [comment]
@Shizheng-Wen Based on your updated configuration, I am able to reproduce the accuracy reported in the paper. 

``` log
--- Final Metrics (Drivaernet Suite - Full Dataset) ---
MSE (x10^-2):       4.2889
MAE (x10^-1):       1.0886
RMSE:               0.1989
Max_Error:          9.6271
Rel L2 Error (%):   20.0844
Rel L1 Error (%):   18.2807
```

Thank you for your patient and detailed responses.
